### PR TITLE
Feature/add parameter in get transactions

### PR DESCRIPTION
**Pull request by Hectormdev**
# Description

This PR adds a new filter in the getTransactions method. Is the first step in order to allow multifiltering in the library. The motivation behind this change is the need of a way of searching for a transaction we dont have the id for.

## Type of change

- [X] New feature 

# How has this PR been tested?

Two tests have been added. One fake that must return an error (not found) and one where it needs to exist (its skipped right now as we dont have an example transaction)

Also snapshots have been updated.

### CODE REVIEW

File: src/Arcus.ts [hunk 1]
```diff
@@ -163,10 +164,11 @@ export class Arcus {
       .then((data) => data.billers);
   }
 
-  getTransactions(): Promise<ArcusTransaction[]> {
+  getTransactions(params?: ArcusGetTransactionParams): Promise<ArcusTransaction[]> {
     const path = '/transactions';
+    const query = params?.externalId ? `/q[external_id_eq]=${params.externalId}` : ``;
```
Comment (DanielRamosAcosta) [inline src/Arcus.ts]: We should use the [`querystring` ](https://nodejs.org/api/querystring.html)native node package here.

File: src/Arcus.spec.ts [hunk 2]
```diff
@@ -64,14 +67,35 @@ describe('Arcus', () => {
...
+      const transactions = await arcus.getTransactions();
 
-    expect(transactions).toHaveLengthWithin(95, 105);
+      expect(transactions).toHaveLengthWithin(95, 105);
 
-    const first = transactions[0];
+      const first = transactions[0];
 
-    expect(first).toMatchSnapshot();
+      expect(first).toMatchSnapshot();
```
Comment (DanielRamosAcosta) [inline src/Arcus.spec.ts]: This tests is probably very flaky, and it might break as soon as we have a new transaction, right?

**Review by DanielRamosAcosta — CHANGES_REQUESTED** [review]

File: src/__snapshots__/Arcus.spec.ts.snap [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 exports[`Arcus getAccount 1`] = `
 Object {
-  "balance": 9982715.67,
+  "balance": 9975465.47,
```
Comment (DanielRamosAcosta) [inline src/__snapshots__/Arcus.spec.ts.snap]: This tests is very flaky, better use "to be within" expectiation

File: src/Arcus.spec.ts [hunk 4]
```diff
@@ -64,14 +67,35 @@ describe('Arcus', () => {
...
+
+    it('getTransaction With paramater and returns error if its not found', async () => {
+      await expect(
+        arcus.getTransactions({
+          externalId: EXAMPLE_FAKE_EXTERNAL_TRANSACTION_ID,
+        }),
+      ).rejects.toBeInstanceOf(ArcusError);
+    });
+
+    it.skip('getTransaction With paramater and returns result', async () => {
```
Comment (DanielRamosAcosta) [inline src/Arcus.spec.ts]: why is this test ignored @Hectormdev ?

File: src/Arcus.spec.ts (hunk 2 above)
Comment (Hectormdev) [reply src/Arcus.spec.ts]: Ill change tests to not use snapshots as they dont seem right for dynamic data, althought this wouldn't break as is the first transaction that has been made what im looking at.

File: src/Arcus.ts (hunk 1 above)
Comment (Hectormdev) [reply src/Arcus.ts]: included!

**Review by DanielRamosAcosta — APPROVED** [review]

**Pull request merged**